File: app/src/main/java/com/example/push/InstanceIdService.java
package com.example.push;

import android.support.annotation.NonNull;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

import com.example.logger.CLog;

public class InstanceIdService extends FirebaseInstanceIdService {
	@Override
	public void onTokenRefresh() {
		String refreshedToken = FirebaseInstanceId.getInstance().getToken();

		if (refreshedToken == null || refreshedToken.isEmpty()) {
			return;
		}

		sendRegistrationToServer(refreshedToken);
	}

	private void sendRegistrationToServer(@NonNull String refreshedToken) {
		CLog.d("( FIREBASE -> onTokenRefresh ) ", refreshedToken);
	}
}

File: app/src/main/kotlin/br/com/vitorsalgado/example/services/push/InstanceIdService.kt
package br.com.vitorsalgado.example.services.push

import br.com.vitorsalgado.example.utils.LogUtility
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService

class InstanceIdService : FirebaseInstanceIdService() {
  override fun onTokenRefresh() {
    val refreshedToken = FirebaseInstanceId.getInstance().token

    if (refreshedToken == null || refreshedToken.isEmpty()) {
      return
    }

    sendRegistrationToServer(refreshedToken)
  }

  private fun sendRegistrationToServer(refreshedToken: String) {
    LogUtility.d("( FIREBASE -> onTokenRefresh ) ", refreshedToken)
  }
}
